### ISSUE TITLE
The FlaskClient.open(test_client) parameter could be explict

### ISSUE THREAD

**Issue by geekKeen** [pallets/flask#2494]
Hi,
I'm always confuse about  keyword args in test method(GET, POST...). 
For example, test post files or json, but I find solutions in StackOverFlow everytimes. 

I debug in source code, find  `FlaskClient.open` and `werkzeug.EnvironBuilder` .

Maybe, we could give some test demo in  Testing and  give more explict description in `FakeClient.open` like [requests](http://docs.python-requests.org/en/master/api/)

**Comment by wgwz** [comment]
Hi @geekKeen. It's a bit unclear what your suggested improvement is. Can you show some examples in the docs that aren't clear? Or some specific testing code you are using which you are happy with (something that ought to be documented)?

**Comment by geekKeen** [comment]
@wgwz , I tryed to write some test case  about  `Flask.test_client()`  to check the view whether is normal , so I want to get more effective infomation  about `Flask.test_client()`, just like posting files or json, custom headers, allow_redirect, response and so on. 
I looked through the document, can't find some snippets about this. I found that `Flask.test_client()` usually call `FlaskClient.open`, I also confused the parameter information about 'FlaskClient.open', now I also confuse about kwargs like `as_tuple` and other kwargs.

There are my suggestions:
1. Give some more effective params infomation about `FlaskClient.open`, 
2. Add some common test case like post files or json, custom request headers, and other function
3. Maybe give information about `FlaskClient.open` response?

\#  I'm so sorry
